src/phoenix/core/traces.py

```diff
@@ -105,18 +105,197 @@ def __hash__(self) -> int:
         return id(self)
 
 
-ParentSpanID: TypeAlias = SpanID
-ChildSpanID: TypeAlias = SpanID
+_ParentSpanID: TypeAlias = SpanID
+_ChildSpanID: TypeAlias = SpanID
+_ProjectName: TypeAlias = str
+
+_DEFAULT_PROJECT_NAME: str = "default"
+
+_SpanItem = Tuple[otlp.Span, _ProjectName]
 
 
 class Traces:
     def __init__(self) -> None:
-        self._queue: "SimpleQueue[Optional[otlp.Span]]" = SimpleQueue()
+        self._span_queue: "SimpleQueue[Optional[_SpanItem]]" = SimpleQueue()
         # Putting `None` as the sentinel value for queue termination.
-        weakref.finalize(self, self._queue.put, END_OF_QUEUE)
+        weakref.finalize(self, self._span_queue.put, END_OF_QUEUE)
+        self._lock = RLock()
+        self._projects: DefaultDict[_ProjectName, "Project"] = defaultdict(
+            Project,
+            {_DEFAULT_PROJECT_NAME: Project()},
+        )
+        self._start_consumers()
+
+    def get_projects(self) -> Iterator[Tuple[str, "Project"]]:
+        with self._lock:
+            projects = tuple(self._projects.items())
+        yield from projects
+
+    def get_trace(self, trace_id: TraceID) -> Iterator[Span]:
+        with self._lock:
+            if not (project := self._projects.get(_DEFAULT_PROJECT_NAME)):
+                return
+        yield from project.get_trace(trace_id)
+
+    def get_spans(
+        self,
+        start_time: Optional[datetime] = None,
+        stop_time: Optional[datetime] = None,
+        root_spans_only: Optional[bool] = False,
+        span_ids: Optional[Iterable[SpanID]] = None,
+    ) -> Iterator[Span]:
+        with self._lock:
+            if not (project := self._projects.get(_DEFAULT_PROJECT_NAME)):
```

Comment:
you are missing project as a parameter here now. Is it because you want to deprecate this?

Reply:
yes. my understanding from yesterday is we're deprecating the corresponding gql query away from `schema` and into `Project`